pkgs/top-level/all-packages.nix

```diff
@@ -16918,7 +16918,7 @@ in
 
     zenpower = callPackage ../os-specific/linux/zenpower { };
 
-    inherit (callPackage ../os-specific/linux/zfs {
+    inherit (callPackages ../os-specific/linux/zfs {
```

Comment:
unrelated to PR, but what's the difference between callPackage and callPackages

I could only find https://github.com/NixOS/nixpkgs/issues/36354 and what's in the source:
```
callPackages = lib.callPackagesWith splicedPackagesWithXorg;
```
but that still doesn't explain what's going on.

Reply:
`callPackage <file>` is for when the file looks like
```nix
{ stdenv }: stdenv.mkDerivation { ... }
```

It makes the result of the call contain a `.override` attribute you can use to change the arguments.

`callPackages <file>` is for when the file looks like
```nix
{ stdenv }: {
  foo = stdenv.mkDerivation { ... };
  bar = stdenv.mkDerivation { ... };
}
```

It makes all the _attributes_ of the result contain a `.override` attribute you can use to change the arguments for any package individually.

If `callPackage` is used on the latter, you'd get
```nix
{
  foo = ...;
  bar = ...;
  override = ...;
}
```
as a result, without any `.override` on the actual derivations, which is why you couldn't `.override` on `pkgs.zfs` before this change.

Reply:
ah, that makes sense. I feel like this should be documented somewhere more formally.

Thank you for the explanation.